[DO NOT MERGE] Update partnership docs (minor fixes) -- Do not merge until Sept 8 (#120)
hashicorp/terraform-docs-common: website/docs/docs/partnerships.mdx

- **CI/CD:** These partners focus on continuous integration and continuous delivery/deployment.
- **VCS:** These partners focus on tracking and managing software code changes.

> Most workflow partners integrate with the Terraform workflow itself. Run tasks allow Terraform Cloud to execute tasks in external systems at specific points in the Terraform Cloud run lifecycle. This offers much more extensibility to Terraform Cloud customers, enabling them to integrate your services into the Terraform Cloud workflow. This feature allows users to add and execute these tasks during the post-plan and the new pre-plan (public beta) stages. Eventually, we will open the third phase of this workflow to Terraform Cloud users, the post apply stage. Reference the [Terraform Cloud Integrations](/docs/partnerships#terraform-cloud-integrations) or [Terraform Enterprise Integrations](/docs/partnerships#terraform-enterprise-integrations) documentation for more details. 

Reviewer: A couple of thoughts on this. First, should we link to the actual run tasks page so folks can get more details about what run tasks are? Also I'm just thinking about how we can make things flow and make it more concise. Oh and finally --> we really try to stay away from future-facing promises in docs. Just because things change, and we've been burned by this before. Instead, i think we should just tell folks the specific points that TFC users can currently choose to execute run tasks.
Thoughts? 

```suggestion
Most workflow partners create [run tasks](/cloud-docs/integrations/run-tasks) that let them integrate their services directly into the Terraform workflow. [Run tasks](/cloud-docs/integrations/run-tasks) let Terraform Cloud execute tasks in external systems at certain points in the Terraform Cloud run lifecycle. Specifically, Terraform Cloud users can configure run tasks to execute during the post-plan and the pre-plan (public beta) run stages. Refer to the [Terraform Cloud Integrations](/docs/partnerships#terraform-cloud-integrations) or [Terraform Enterprise Integrations](/docs/partnerships#terraform-enterprise-integrations) documentation for more details. 
```

Author: Yup, sounds good!

e) Contributing to Terraform [guidelines](https://github.com/hashicorp/terraform/blob/master/.github/CONTRIBUTING.md)

f) HashiCorp Developer Discuss [forum](https://discuss.hashicorp.com/c/terraform-providers/tf-plugin-sdk/43)

Please submit questions or suggestions about the Terraform SDK and provider development to the HashiCorp Terraform plugin SDK forum. If you are new to provider development and would like assistance, you can leverage one of the following development agencies that have developed Terraform providers in the past:

| Partner            | Email                       | Website                                             |
| ------------------ | :-------------------------- | :-------------------------------------------------- |
| Crest Data Systems | <malhar@crestdatasys.com>   | [www.crestdatasys.com](http://www.crestdatasys.com) |

> | Tech Mahindra      | <hashicorp@digitalonus.com> | [www.digitalonus.com](http://www.digitalonus.com)   |

Reviewer: @gautambaghel do we need to update the email and website too?

Author: Unfortunately no @laurapacilio, that remains the same. It's odd.

Reviewer: huh, okay then! Just checking :)

### 5. Support

> Getting a new provider built and published to the Terraform Registry is just the first step towards enabling your users with a quality Terraform integration. Once a partner's provider has been published, on-going effort is required to maintain the provider.

Reviewer: ```suggestion
Building and publishing a new provider to the Terraform Registry is just the first step toward a quality Terraform integration. Integrations require consistent maintenance throughout their lifecycle.
```

> HashiCorp Terraform has an extremely wide community of users and contributors and we encourage everyone to report issues however small, as well as help resolve them when possible. We expect that all partner provider publishers will continue to maintain the provider and address any issues users report in a timely manner. This includes resolving all critical issues within 48 hours and all other issues within 5 business days. HashiCorp reserves the right to remove partner status from any integration that is no longer being maintained.

Reviewer: I was thinking that "is not being maintained" is kind of vague. What if we said "that doesn't meet our maintenance requirements" --> Thought is someone could maybe argue they ARE maintaining it....but if it's not up to our expectations, we still can pull it. Thoughts?

```suggestion
HashiCorp Terraform has an extremely wide community of users and contributors, and we encourage the community to report issues and help resolve them. However, we expect that all partner provider publishers continue maintaining the provider and that they address user-reported issues in a timely manner. This includes resolving all critical issues within 48 hours and all other issues within 5 business days. HashiCorp reserves the right to remove partner status from any integration that does not meet our maintenance requirements.
```

Author: I think so, makes sense.

> Vendors who choose not to support their provider and prefer to make it a community-supported provider will no longer be listed as Partner.

Reviewer: Should they notify us if they want to make this switch?

Author: Not really, nobody says I'm losing faith in you and no longer want to support this. There's an issue with deprecation in the industry overall

Reviewer: huh also interesting!

## Terraform Cloud Integrations

Run tasks allow Terraform Cloud to execute tasks in external systems at specific points in the Terraform Cloud run lifecycle. It allows users to add and execute these tasks during the pre-plan (public beta) and post-plan stages of the Terraform run. Run tasks are executed by sending an API payload to the external system. This payload contains a collection of run-related information and a callback URL which the external system can use to send updates back to Terraform Cloud.

The external system can then use this run information and respond back to Terraform Cloud with a passed or failed status. Terraform Cloud uses this status response to determine if a run should proceed, based on the task's enforcement settings within a workspace.

Partners who successfully complete the Terraform Cloud integration checklist obtain a Terraform Cloud badge. This signifies HashiCorp has verified the integration and the partner is a member of the HashiCorp Technology Partner Program.

> - Note: As of September 2022, pre-plan run tasks are available only as a public beta feature, are subject to change, and not all customers will see this functionality in their Terraform Cloud organization since this is currently enabled by default for our business tier customers of Terraform Cloud. If you have a customer that is interested in run tasks and are not a current Terraform Cloud for Business customer please request them to [sign up using the form](https://docs.google.com/forms/d/e/1FAIpQLSf3JJIkU05bKWov2wXa9c-QV524WNaHuGIk7xjHnwl5ceGw2A/viewform).

Reviewer: This form is no longer available :) Should we remove the note about signing up for the beta?

```suggestion
- Note: As of September 2022, pre-plan run tasks are available only as a public beta feature, are subject to change, and not all customers will see this functionality in their Terraform Cloud organization. This feature is enabled by default for our business tier customers of Terraform Cloud. If you have a customer that is interested in run tasks and are not a current Terraform Cloud for Business customer, please ask them to [sign up for the public beta](https://docs.google.com/forms/d/e/1FAIpQLSf3JJIkU05bKWov2wXa9c-QV524WNaHuGIk7xjHnwl5ceGw2A/viewform).
```

### 2. Develop & Test

> Partners should build an integration using the [Run task APIs in Terraform Cloud](/cloud-docs/api-docs/run-tasks). For example, [Snyk](https://docs.snyk.io/features/integrations/ci-cd-integrations/integrating-snyk-with-terraform-cloud) created a run tasks to automatically detect misconfigurations in the Terraform plan. To learn more about how post-plan run tasks can enhance the Terraform Cloud workflow, refer to the following diagram and read our [run tasks announcement](https://www.hashicorp.com/blog/terraform-cloud-run-tasks-beta-now-available). 

Reviewer: ```suggestion
Partners should build an integration using the [Run task APIs in Terraform Cloud](/cloud-docs/api-docs/run-tasks). For example, [Snyk](https://docs.snyk.io/features/integrations/ci-cd-integrations/integrating-snyk-with-terraform-cloud) created a run task to automatically detect misconfigurations in the Terraform plan. To learn more about how post-plan run tasks can enhance the Terraform Cloud workflow, refer to the following diagram and read our [run tasks announcement](https://www.hashicorp.com/blog/terraform-cloud-run-tasks-beta-now-available). 
```

> Partners should build an integration using [Run task APIs in Terraform Cloud](/cloud-docs/api-docs/run-tasks). To better understand how post-plan run task enhances the workflow, see diagram listed below and check out our [announcement about Terraform run task](https://www.hashicorp.com/blog/terraform-cloud-run-tasks-beta-now-available). [Snyk](https://docs.snyk.io/features/integrations/ci-cd-integrations/integrating-snyk-with-terraform-cloud), for example, created an integration to detect configuration anomalies in code while reducing risk to the infrastructure. For additional API resources, [click here](/cloud-docs/api-docs).

Reviewer: Do we need to link them to the additional API stuff if we already link to the Run Task APIs above?

```suggestion
Partners should build an integration using the [Run task APIs in Terraform Cloud](/cloud-docs/api-docs/run-tasks). For example, [Snyk](https://docs.snyk.io/features/integrations/ci-cd-integrations/integrating-snyk-with-terraform-cloud) created a run tasks to automatically detect misconfigurations in the Terraform plan. To learn more about how post-plan run tasks can enhance the Terraform Cloud workflow, refer to the following diagram and read our [run tasks announcement](https://www.hashicorp.com/blog/terraform-cloud-run-tasks-beta-now-available). 
```

> **Currently, pre-plan is the only available as public beta. [Updated September 1, 2022]**

Reviewer: We typically use notes for this type of information. Let me know if you don't like this!

```suggestion
-> **Note:** Pre-plan run tasks are only available as a public beta. [Updated September 1, 2022]
```

<img src="/img/docs/tfe-badge.png" alt="Terraform Cloud Partnership Badge" style={{display:'block', margin:'auto', width:'200px', height:'200px'}}/>

-> Contact us at [technologypartners@hashicorp.com](mailto:technologypartners@hashicorp.com) with any questions or feedback.

> Upon successfully building Run Tasks integration with Terraform Cloud partners can reach out to their Alliance Manager to validate it against Terraform Enterprise. It’s common for customers to deploy Terraform Enterprise in a private cloud or in air-gapped environment in which case partners need ensure that their product can communicate bi-directionally with TFE inside the customer’s network.

Reviewer: ```suggestion
After successfully building a Run Tasks integration with Terraform Cloud, partners should contact their Alliance Manager to validate the integration against Terraform Enterprise. Many customers deploy Terraform Enterprise in a private cloud or air-gapped environment, so partners must ensure that their product can communicate bi-directionally with Terraform Enterprise inside these types of networks.
```
